components/Pagination.js

```diff
@@ -0,0 +1,145 @@
+import { useEffect, useState } from "react";
+import { FaChevronLeft, FaChevronRight } from "react-icons/fa";
+import { BsChevronDoubleLeft, BsChevronDoubleRight } from "react-icons/bs";
+
+const Pagination = ({
+  totalUsers,
```

Comment:
I think this should be kept generic, so we can reuse the component. For example `data`